ntp-admin/src/http_entrypoints.rs

```diff
@@ -0,0 +1,127 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::context::ServerContext;
+use dropshot::HttpError;
+use dropshot::HttpResponseOk;
+use dropshot::RequestContext;
+use ntp_admin_api::*;
+use ntp_admin_types::TimeSync;
+use slog::error;
+use slog::info;
+use slog_error_chain::InlineErrorChain;
+use std::net::IpAddr;
+use std::net::Ipv6Addr;
+use std::str::FromStr;
+use std::sync::Arc;
+
+type NtpApiDescription = dropshot::ApiDescription<Arc<ServerContext>>;
+
+pub fn api() -> NtpApiDescription {
+    ntp_admin_api_mod::api_description::<NtpAdminImpl>()
+        .expect("registered entrypoints")
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum TimeSyncError {
+    #[error("failed to execute chronyc within NTP zone")]
+    ExecuteChronyc(#[source] std::io::Error),
+    #[error(
+        "failed to parse chronyc tracking output: {reason} (stdout: {stdout:?})"
+    )]
+    FailedToParse { reason: &'static str, stdout: String },
+}
+
+impl From<TimeSyncError> for HttpError {
+    fn from(err: TimeSyncError) -> Self {
+        // All errors are currently treated as 500s
+        HttpError::for_internal_error(InlineErrorChain::new(&err).to_string())
+    }
+}
+
+enum NtpAdminImpl {}
+
+impl NtpAdminImpl {
+    async fn timesync_get(
+        ctx: &ServerContext,
+    ) -> Result<TimeSync, TimeSyncError> {
+        let log = ctx.log();
+        info!(log, "querying chronyc");
+
+        let output = tokio::process::Command::new("/usr/bin/chronyc")
+            .args(["-c", "tracking"])
+            .output()
+            .await
+            .map_err(TimeSyncError::ExecuteChronyc)?;
+        let stdout = String::from_utf8_lossy(&output.stdout).to_string();
+
+        let v: Vec<&str> = stdout.split(',').collect();
+
+        if v.len() < 10 {
+            error!(log, "too few fields");
+            return Err(TimeSyncError::FailedToParse {
+                reason: "too few fields",
+                stdout,
+            });
+        }
+
+        let Ok(ref_id) = u32::from_str_radix(v[0], 16) else {
+            error!(log, "failed to parse ref_id");
```

Comment:
would it be worthwhile to include the invalid value here (and subsequently)?

Reply:
Updated!